src/utils/event-manager.js

```diff
@@ -0,0 +1,137 @@
+import {createGestureManager, EVENT_RECOGNIZER_MAP} from './gesture-manager';
+
+const EVENT_HANDLER_MAP = {
+  pointerdown: 'mousedown',
+  pointermove: 'mousemove',
+  pointerup: 'mouseup',
+  mousedown: 'mousedown',
+  mousemove: 'mousemove',
+  mouseup: 'mouseup'
+};
+const GESTURE_ALIASES = {
+  click: 'tap'
+};
+
+const DEFAULT_OPTIONS = {
+  handleRawEventsWithGestureManager: true
+};
+
+/**
+ * Single API for subscribing to events about both "raw" input (e.g. 'mousemove', 'click')
+ * and gestural input (e.g. 'panstart', 'tap').
+ */
+class EventManager {
+  constructor(element, options) {
+    this.element = element;
+    this.handlers = {};
+    this.gestureManager = createGestureManager(element, options);
+
+    // If specified, use GestureManager for raw event handling
+    // as well as for gesture recognition.
+    if (options.handleRawEventsWithGestureManager) {
+      this._onRawInput = this._onRawInput.bind(this);
+      this.gestureManager.startHandlingRawEvents(this._onRawInput);
+    }
+  }
+
+  on(event, handler) {
```

Comment:
It's the `hammer.input` thing -- we need to manage our own handlers because hammer doesn't maintain a one-to-one mapping of `eventName: eventHandler` for the events we pull out of `hammer.input` events. See: `_onRawInput`.